public/app/features/datasources/components/picker/DataSourceList.tsx

```diff
@@ -48,18 +48,33 @@ export function DataSourceList(props: DataSourceListProps) {
     variables: props.variables,
   });
 
+  const { recentlyUsedDataSources, pushRecentlyUsedDataSource } = useRecentlyUsedDataSources();
+
+  const orderedDataSources = orderDataSourcesByRecentlyUsed(recentlyUsedDataSources, dataSources);
```

Comment:
Here, we should also move the current selected DS to the top. 

The order should be:
- Current
- Recently used
- Variables
- Rest is sorted alphabetically.

So I'd rename this to `sortDataSources(...)` and delegate everything to that function. WDYT?

Reply:
Sounds good. I'll implement it all in that function.